pkg/config/node.go

```diff
@@ -278,8 +278,21 @@ func IsUpgradeStillRunning(kubeconfigPath string) (bool, error) {
 		return false, err
 	}
 
+	kubeletVersion := ""
 	// Go to all node types identified in GetNodes()
 	for nodeRole := range nodes {
+		// Verify kubelet versions match. In EUS upgrades we may end up in an
+		// intermediate state where all of the nodes are "updated" as far as
+		// MCO is concerned, but are actually on different versions of OCP.
+		// In those cases, we do not consider the upgrade complete because not
+		// all nodes are ready for migration.
+		if kubeletVersion == "" {
+			kubeletVersion = nodes[nodeRole][0].Status.NodeInfo.KubeletVersion
+		}
+		if kubeletVersion != nodes[nodeRole][0].Status.NodeInfo.KubeletVersion {
```

Comment:
I might be wrong but if you are setting kubeletVersion in line 290 how kubeletVersion will be different from nodes[NodeRole][0]... ? Maybe are we looking for `else if` ?

Reply:
We only set kubeletVersion there the first time through the loop. Once it has a non-empty value we skip the first assignment.

Also maybe worth noting that I'm only checking a single node from each role. That should be fine since if there's any variation within the role it should be handled by the machine-config check.